src/components/DeleteMovieForm/DeleteMovieForm.tsx

```diff
@@ -0,0 +1,31 @@
+import Dialog from "../Dialog/Dialog";
+import {useLoaderData, useNavigate} from "react-router-dom";
+import {DialogType} from "../../constants/DialogType";
+import axios from "axios";
+import DeleteDialog from "../DeleteDialog/DeleteDialog";
+
+async function deleteMovie(id: number) {
+    await axios.delete(`http://localhost:4000/movies/${id}`);
+}
+
+function DeleteMovieForm() {
+    const navigate = useNavigate();
+    const {movieDetails} = useLoaderData();
+
+    async function handleDeleteMovie(id: number) {
+        try {
+            await deleteMovie(id);
+            navigate(`/`, { replace: true });
+        } catch (error) {
+            console.error("Error deleting a movie: ", error);
+        }
+    }
+
+    return <Dialog dialogTitle={DialogType.DeleteMovie}
+                   content={<DeleteDialog movieToDelete={movieDetails}
```

Comment:
I did search => edit the movie => click back to search page and i as return again to edit modal. the issue with routing here
![image](https://github.com/user-attachments/assets/8298784a-2487-45df-bbfd-bfd7736961d5)